app/components/conferences/admin/SponsorForm.js

```diff
@@ -0,0 +1,365 @@
+                            "starts-at": event.data.attributes["starts-at"],
+                            "ends-at": event.data.attributes["ends-at"],
+                            "original-image-url": event.data.attributes["original-image-url"],
+                            "logo-url": event.data.attributes["logo-url"],
+                            timezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+                            "is-sponsors-enabled": true
+                        },
+                        id: event.data.id,
+                        type: "event"
+                    }
+                }
+
+                await editEvent(eventData, token, event.data.id)
+            }
+
+            const res = await publishSponsor(data, token);
+            setSponsors([...sponsors,res.data.data])
+            setModalShow(false)
+        } catch (error) {
+            console.log("Error in publishing sponsor", error)
+        }
+    }
+
+    const handleSubmit = async () => {
+        try {
+            let token = Cookies.get("event_auth");
+            token
+                ? (token = JSON.parse(token).access_token)
+                : new Error("Please, Sign in again");
+
+            sponsors.map((value) => {
+                updateSponsor({data : value},token);
+            })
+
+            handleToast({ show: true, msg: "Sponsor Updated Successfully" })
+            router.push(`/conferences/admin/c/${event.data.id}/sessions`)
+        } catch (error) {
+            console.log("Error in updating sponsors",error)   
+        }
+    }
+
+    useEffect(() => {
+        const sponsorInfo = async () => {
+            try {
+                let token = Cookies.get("event_auth");
+                token
+                    ? (token = JSON.parse(token).access_token)
+                    : new Error("Please, Sign in again");
+
+                const res = await getSponsorsDetails(event.data.id, token)
+                setSponsors(res.data.data)
+            } catch (e) {
+                console.error("An error occurred while fetching Sponsors", e);
+            }
+        };
+        sponsorInfo();
+    }, [])
+
+    return (
+        <div className="m-3">
+            <Stack className={styles.speaker_add_btn}>
+                <Button onClick={() => { setModalShow(true) }}>Add Sponsors</Button>
+            </Stack>
+            <SponsorModal
+                show={modalShow}
+                onHide={() => setModalShow(false)}
+                handleAddSponsor={handleAddSponsor}
+                modalHandleChange={modalHandleChange}
+                load={load}
+            />
+            <SponsorList
+                sponsors={sponsors}
+                handleChange={handleChange}
+                handleDelete={handleDelete}
+            />
+            <Button variant="primary" type="submit" onClick={handleSubmit} className="my-2">
+                Next
+            </Button>
```

Comment:
If the user clicks on the "Next" button on the Basic detail form, the event is supposed to be saved as draft, so we need an "Publish" button (if the event is in draft state)

Reply:
I have added the Publish Event Button if the event is in the draft state